### PR TITLE
Icon table with metadata

### PR DESCRIPTION
**Pull request by rajsite**
# Pull Request

## 🤨 Rationale

Create a table for icons in storybook that shows the tag names and metaphors.

## 👩‍💻 Implementation

- Rendered the icons using the nimble-table
- Show metaphors in a separate column that is not sortable
- Set the height of the table large enough so that all the rows render in the DOM. That way the page is searchable with normal ctrl+f search and there are no nested scrollbars. Can also use normal browser zoom to see the icons bigger.

## 🧪 Testing

Manually checked in storybook

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/icon-base/tests/icons.stories.ts [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
 } from '../../theme-provider/design-token-names';
+import { Table, tableTag } from '../../table';
+import { tableColumnIconTag } from '../../table-column/icon';
+import { mappingIconTag } from '../../mapping/icon';
+import { tableColumnTextTag } from '../../table-column/text';
+import { iconMetadata } from './icon-metadata';
 
-const nimbleIconComponents = Object.values(nimbleIconComponentsMap);
+type IconName = keyof typeof nimbleIconComponentsMap;
+const data = Object.values(nimbleIconComponentsMap).map(klass => ({
```
Comment (fredvisser) [inline packages/nimble-components/src/icon-base/tests/icons.stories.ts:21]: Just curious `klass` is an unreserved keyword version of `class`? Or something else?

**Review by fredvisser — APPROVED** [review]

File: packages/nimble-components/src/icon-base/tests/icons.stories.ts (hunk 1 above)
Comment (rajsite) [reply packages/nimble-components/src/icon-base/tests/icons.stories.ts]: yep 😅 something deep inside my finger memory, but I wouldn't let that through review these days so good catch, Updated to a normal name of iconClass

File: packages/nimble-components/src/icon-base/tests/icons.stories.ts [hunk 2]
```diff
@@ -1,22 +1,33 @@
...
+}));
+
+type Data = (typeof data)[number];
 
 interface IconArgs {
     severity: keyof typeof IconSeverity;
+    tableRef: Table<Data>;
 }
 
 const metadata: Meta<IconArgs> = {
```
Comment (jattasNI) [inline packages/nimble-components/src/icon-base/tests/icons.stories.ts:33]: I was finding it slightly jarring to see a huge table with no context. How about a bit of docs above it like this?
```suggestion
const metadata: Meta<IconArgs> = {
    description: 'Nimble icons can be slotted into other components or used independently. Each icon is available as a custom element. For example, `<nimble-icon-add></nimble-icon-add>`'.
```

**Review by jattasNI — APPROVED** [review]

File: packages/nimble-components/src/icon-base/tests/icons.stories.ts [hunk 3]
```diff
@@ -52,26 +47,57 @@ Set severity on the element to switch between the theme-aware color options.
...
     render: createUserSelectedThemeStory(html`
-        <style class="code-hide">
-            .container > * {
-                padding: 5px;
-            }
-        </style>
-        <div class="container">
-            ${repeat(() => nimbleIconComponents, iconTemplate)}
-        </div>
+        <${tableTag}
```
[the remark this answers is not in the document]
Comment (rajsite) [reply packages/nimble-components/src/icon-base/tests/icons.stories.ts:89]: Disabled show code for icons and the tokens stories via `{ parameters: { docs: { source: { code: null } } } }` similar to the matrix stories

**Pull request merged**